SuperblocksHQ/ethereum-studio
# Change EVM transaction value data from integer to big number

### Description of the Change

_EVM_ transaction values are stored as integer, where max integer is `2ˆ53`. Given `1000000000000000000 > 9007199254740992`, report #145 shows to be valid.

This request aims to address the issue by converting the inner _Buffer_ to big number (_BN_) instead. 

Log:

* Changes value in `getBlock`, `getTransaction` and `getTransactionByBlock` to
transform transaction value from _Buffer_ to _BN_ (big number) instead of _integer_
* Bump _evm iframe_ version
* Update _evm_ `README`


### Benefits

1. Ability to call payable function with large numbers (#145)
2. Ability to deploy payable contract with large value (#146)

### Possible Drawbacks

Unknown. 
_Note: regenerated distributable file._

### Verification Process

**a. Ability to call payable function with large numbers (#145)**

From scratch (no _node_modules_):
```
npx lerna bootstrap --hoist && npm start
```

_Localhost_:
[![Edit Project](https://studio.ethereum.org/static/img/open-studio.svg)](http://localhost:4000/5dcd34ef26b473001233e0ae)

1. _Compile_
2. _Deploy_ to _Browser_ network
3. Open _Interact_ panel
4. Set `update` function _parameter_ to `test`
5. Set `update` function _Value_ to `90000000000000000000` Wei
6. Press `update` button
7. Verify `message` contents changed from `Hello World!` to `test`
8. Verify browser console output shows no errors
9. Verify _Account balance_ changed from 100 Ether to ~9.99 eth

**2. Ability to deploy payable contract with large value (#146)**

From _Localhost_, with patch #146:
[![Edit Project](https://studio.ethereum.org/static/img/open-studio.svg)](http://localhost:4000/5dcd34ef26b473001233e0ae)

1. Change the `constructor` to:
```
    constructor() public payable {
        require(msg.value == 90000000000000000000 wei, "90000000000000000000 wei initial funding required");
        value = msg.value;
    }
```
2. _Configure_
3. Locate `Value` input and add value of `90000000000000000000`, then _Save_
4. _Compile_
5. _Deploy_
6. Go to _Interact_ panel and check if value constant reads `90000000000000000000`
7. _Account balance_ is expected to read ~9.9 eth




### Github Issues

* Resolves #145 
* Unblocks #146
